encfs/Context.cpp

```diff
@@ -75,15 +81,39 @@ void EncFS_Context::setRoot(const std::shared_ptr<DirNode> &r) {
   }
 }
 
-bool EncFS_Context::isMounted() { return root != nullptr; }
-
-void EncFS_Context::getAndResetUsageCounter(int *usage, int *openCount) {
+bool EncFS_Context::usageAndUnmount(int timeoutCycles) {
   Lock lock(contextMutex);
 
-  *usage = usageCount;
-  usageCount = 0;
+  if (root != nullptr) {
 
-  *openCount = openFiles.size();
+    if (usageCount == 0) {
+      ++idleCount;
+    }
+    else {
+      idleCount = 0;
+    }
+    VLOG(1) << "idle cycle count: " << idleCount << ", timeout at "
+            << timeoutCycles;
+
+    usageCount = 0;
+
+    if (idleCount >= timeoutCycles) {
+      if (!openFiles.empty()) {
+        if (idleCount % timeoutCycles == 0) {
+          RLOG(WARNING) << "Filesystem inactive, but " << openFiles.size()
+                        << " files opened: " << this->opts->mountPoint;
+        }
```

Comment:
"return false" to get rid of the else. The if chain is three levels deep and hard to follow

Reply:
and done !
Thx for your review 👍